packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/tags/CreateTagsTests.java

```diff
@@ -0,0 +1,218 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.rest.tags;
+
+import static org.alfresco.utility.data.RandomData.getRandomName;
+import static org.alfresco.utility.report.log.Step.STEP;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.CONFLICT;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.OK;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.alfresco.rest.RestTest;
+import org.alfresco.rest.model.RestTagModel;
+import org.alfresco.rest.model.RestTagModelsCollection;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class CreateTagsTests extends RestTest
+{
+    private static final String FIELD_ID = "id";
+    private static final String FIELD_TAG = "tag";
+    private static final String FIELD_COUNT = "count";
+    private static final String TAG_NAME_PREFIX = "tag-name";
+
+    private UserModel admin;
+    private UserModel user;
+
+    @BeforeClass
+    public void init()
+    {
+        admin = dataUser.getAdminUser();
+        user = dataUser.createRandomTestUser();
+    }
+
+    /**
+     * Verify if tag does not exist in the system, create one as admin and check if now it's there.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS })
+    public void testCreateSingleTag()
+    {
+        STEP("Create single tag as admin");
+        final RestTagModel tagModel = createTagModelWithRandomName();
+        final RestTagModel createdTag = restClient.authenticateUser(admin).withCoreAPI().createSingleTag(tagModel);
+
+        restClient.assertStatusCodeIs(CREATED);
+        createdTag.assertThat().field(FIELD_TAG).is(tagModel.getTag())
+            .assertThat().field(FIELD_ID).isNotEmpty();
+
+        STEP("Verify that tag does exist in the system");
+        RestTagModel tag = restClient.authenticateUser(admin).withCoreAPI().getTag(createdTag);
+        restClient.assertStatusCodeIs(OK);
+        tag.assertThat().isEqualTo(createdTag);
+    }
+
+    /**
+     * Create multiple orphan tags.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.TAGS })
+    public void testCreateMultipleTags()
+    {
+        STEP("Create several tags as admin");
+        final List<RestTagModel> tagModels = IntStream.range(0, 3)
+            .mapToObj(i -> createTagModelWithName(getRandomName(TAG_NAME_PREFIX + "-" + i).toLowerCase()))
```

Comment:
Why `toLowerCase`?

Reply:
Tags in repo are all lower case and I had to put it somewhere - here or in assertion 😅

Reply:
Ok. After this I saw that you had another test for upper case, but I didn't come back and tidy my comment.

Reply:
No worries - thanks for mentioning that!